auth/service_test.go

```diff
@@ -0,0 +1,186 @@
+package auth_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/fabric8-services/fabric8-tenant/auth"
+	testsupport "github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/doubles"
+	"github.com/fabric8-services/fabric8-tenant/test/recorder"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+)
+
+func TestResolveUserToken(t *testing.T) {
+	// given
+	authService, r, err := testdoubles.NewAuthClientService("../test/data/token/auth_resolve_target_token", "http://authservice", recorder.WithJWTMatcher)
+	require.NoError(t, err)
+	defer r.Stop()
+	tok, err := testsupport.NewToken(
+		map[string]interface{}{
+			"sub": "user_foo",
+		},
+		"../test/private_key.pem",
+	)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+		// when
+		username, accessToken, err := authService.ResolveUserToken(context.Background(), "some_valid_openshift_resource", tok.Raw)
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, "user_foo", username)
+		assert.Equal(t, "an_openshift_token", accessToken)
+	})
+
+	t.Run("invalid resource", func(t *testing.T) {
+		// when
+		_, _, err := authService.ResolveUserToken(context.Background(), "some_invalid_resource", tok.Raw)
+		// then
+		require.Error(t, err)
+	})
+
+	t.Run("empty access token", func(t *testing.T) {
+		// when
+		_, _, err := authService.ResolveUserToken(context.Background(), "some_valid_openshift_resource", "")
+		// then
+		require.Error(t, err)
+	})
+}
+
+func TestResolveServiceAccountToken(t *testing.T) {
```

Comment:
sub-tests are not calling `ResolveSaToken()` but calling `ResolveTargetToken()` which is anyway sort of cover by `TestResolveUserToken()` test.  why this `TestResolveServiceAccountToken()` test is needed OR can we have test for `ResolveSaToken()` .. at current state of test, I see it's almost duplicate and not adding much value.

Reply:
fixed in 161429